pyquil/_parser/Quil.g4

```diff
@@ -0,0 +1,203 @@
+grammar Quil;
+
+////////////////////
+// PARSER
+////////////////////
+
+quil                : allInstr (NEWLINE+ allInstr)* NEWLINE* EOF ;
+
+allInstr            : defGate
+                    | defCircuit
+                    | instr
+                    ;
+
+instr               : gate
+                    | measure
+                    | defLabel
+                    | halt
+                    | jump
+                    | jumpWhen
+                    | jumpUnless
+                    | resetState
+                    | wait
+                    | classicalUnary
+                    | classicalBinary
+                    | nop
+                    | include
+                    | pragma
+                    ;
+
+// C. Static and Parametric Gates
+
+gate                : name (LPAREN param (COMMA param)* RPAREN)? qubit+ ;
+
+name                : IDENTIFIER ;
+qubit               : UNSIGNED_INT ;
+
+param               : dynamicParam | expression ;
+dynamicParam        : LBRACKET qubit (MINUS qubit)? RBRACKET ;
```

Comment:
I would say that this should be `LBRACKET UNSIGNED_INT (MINUS UNSIGNED_INT)? RBRACKET`

They don't refer to qubits.

Reply:
good point, fixed